Tests/SwiftGenKitTests/TestsHelper.swift

```diff
@@ -159,29 +169,32 @@ private func convertToDictionary(_ value: Any) -> [String: Any]? {
   switch value {
   case let value as [String: Any]:
     return value
-  case let value as NSObject:
-    return convertNSObjectToDictionary(value)
+  case let value as NSObject & ContextConvertible:
+    return value.convertToDictionary()
   default:
     return nil
   }
 }
 
-private func convertNSObjectToDictionary(_ object: NSObject) -> [String: Any] {
-  var result: [(String, Any)] = []
-  var mirror: Mirror? = Mirror(reflecting: object)
-
-  repeat {
-    // At runtime when mirroring, labels for lazy vars get the "$__lazy_storage_$_" prefix
-    // so we need to delete the prefix to handle those lazy var cases.
-    // We use `object.value(forKey:) to force load lazy variables because `someMirrorChild.value`
-    // evaluates to `nil` for lazy variables and doesn't trigger the lazy var evaluation.
-    result += mirror?.children
-      .compactMap { $0.label?.deletingPrefix("$__lazy_storage_$_") }
-      .map { ($0, object.value(forKey: $0) as Any) } ?? []
-    mirror = mirror?.superclassMirror
-  } while mirror != nil
-
-  return Dictionary.init(uniqueKeysWithValues: result)
+extension ContextConvertible where Self: NSObject {
+  func convertToDictionary() -> [String: Any] {
+    var result: [(String, Any)] = []
+    var mirror: Mirror? = Mirror(reflecting: self)
+
+    repeat {
+      // At runtime when mirroring, labels for lazy vars get the "$__lazy_storage_$_" prefix
+      // so we need to delete the prefix to handle those lazy var cases.
+      // We use `object.value(forKey:) to force load lazy variables because `someMirrorChild.value`
+      // evaluates to `nil` for lazy variables and doesn't trigger the lazy var evaluation.
+      result += mirror?.children
+        .compactMap { $0.label?.deletingPrefix("$__lazy_storage_$_") }
+        .filter { !$0.hasPrefix("_") }
```

Comment:
Is this line still needed? You added it in b5f72c466d22ce290372240b1a6ad05ef15777c2

Reply:
Mmmh probably not indeed. Will remove.